specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-10-17-preview/workbookTemplates_API.json

```diff
@@ -0,0 +1,467 @@
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/WorkbookError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookTemplateUpdate": {
+            "$ref": "./examples/WorkbookTemplateUpdate.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "WorkbookTemplateResource": {
+      "properties": {
+        "resourceId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource name."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "WorkbookTemplatesListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/WorkbookTemplate"
+          },
+          "description": "An array of workbook templates."
+        }
+      },
+      "description": "WorkbookTemplate list result."
+    },
+    "WorkbookTemplate": {
+      "description": "An Application Insights workbook template definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/WorkbookTemplateResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a workbook template for an Azure resource.",
+          "$ref": "#/definitions/WorkbookTemplateProperties"
+        }
+      }
+    },
+    "WorkbookTemplateProperties": {
+      "description": "Properties that contain a workbook template.",
+      "required": [
+        "id",
+        "templateData",
+        "galleries"
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id of the workbook template. This is GUID value."
+        },
+        "priority": {
+          "type": "integer",
+          "description": "Priority of the template. Determines which template to open when a workbook gallery is opened in viewer mode."
+        },
+        "author": {
+          "type": "string",
+          "description": "Information about the author of the workbook template."
+        },
+        "templateData": {
```

Comment:
Does this have no schema that can be defined?  How will a user know what to put here?

Reply:
This is a complex schema that supports and drives the Workbook UI. We are going to create experiences through the UI to enable users to create workbook templates from the workbook that they have been working on or through existing templates in our github workbooks repo https://github.com/microsoft/Application-Insights-Workbooks